### PR TITLE
[6.0] Revise liftToConstant method

### LINKED ISSUE #905: [v6.0] Revise liftToConstant method
The method should be constrained in which values it can lift.
https://github.com/ScorexFoundation/sigmastate-interpreter/blob/72baa09432c1e21872bd1034e949dde133fecccf/interpreter/jvm/src/main/scala/sigmastate/Platform.scala#L19

### PR DESCRIPTION
**Pull request by kushti**
Close #905

### CODE REVIEW

File: data/js/src/main/scala/sigma/Platform.scala [hunk 1]
```diff
@@ -14,31 +14,32 @@ object Platform {
...
       case v: Long => Nullable(mkConstant[SLong.type](v, SLong))
-      case v: BigInteger => Nullable(mkConstant[SBigInt.type](CBigInt(v), SBigInt))
+      case v: BigInteger if !VersionContext.current.isV6SoftForkActivated => Nullable(mkConstant[SBigInt.type](CBigInt(v), SBigInt))
       case n: sigma.BigInt => Nullable(mkConstant[SBigInt.type](n, SBigInt))
       case n: sigma.UnsignedBigInt => Nullable(mkConstant[SUnsignedBigInt.type](n, SUnsignedBigInt))
       case ge: GroupElement => Nullable(mkConstant[SGroupElement.type](ge, SGroupElement))
       case b: Boolean => Nullable(if (b) TrueLeaf else FalseLeaf)
-      case v: String => Nullable(mkConstant[SString.type](v, SString))
+      case v: String if !VersionContext.current.isV6SoftForkActivated => Nullable(mkConstant[SString.type](v, SString))
+      case h: Header if VersionContext.current.isV6SoftForkActivated  => Nullable(mkConstant[SHeader.type](h, SHeader))
```
Comment (aslesarenko) [inline data/js/src/main/scala/sigma/Platform.scala:35]: PreHeader is also supported type, they usually come together everywhere.

File: data/jvm/src/main/scala/sigma/Platform.scala [hunk 2]
```diff
@@ -16,29 +16,32 @@ object Platform {
...
       case v: Short => Nullable(mkConstant[SShort.type](v, SShort))
       case v: Int => Nullable(mkConstant[SInt.type](v, SInt))
       case v: Long => Nullable(mkConstant[SLong.type](v, SLong))
-      case v: BigInteger => Nullable(mkConstant[SBigInt.type](SigmaDsl.BigInt(v), SBigInt))
+      case v: BigInteger if !VersionContext.current.isV6SoftForkActivated  => Nullable(mkConstant[SBigInt.type](SigmaDsl.BigInt(v), SBigInt))
       case n: sigma.BigInt => Nullable(mkConstant[SBigInt.type](n, SBigInt))
       case ge: GroupElement => Nullable(mkConstant[SGroupElement.type](ge, SGroupElement))
       case b: Boolean => Nullable(if (b) TrueLeaf else FalseLeaf)
-      case v: String => Nullable(mkConstant[SString.type](v, SString))
+      case h: Header if VersionContext.current.isV6SoftForkActivated  => Nullable(mkConstant[SHeader.type](h, SHeader))
```
Comment (aslesarenko) [inline data/jvm/src/main/scala/sigma/Platform.scala:34]: PreHeader is missing.

File: interpreter/shared/src/test/scala/sigma/ast/SigmaBuilderTest.scala [hunk 3]
```diff
@@ -122,6 +122,13 @@ class SigmaBuilderTest extends AnyPropSpec with ScalaCheckPropertyChecks with Ma
     testSuccess(arr, TransformingSigmaBuilder.mkCollectionConstant(arr, c.tpe))
   }
 
+  def testArrayFailure[T <: SType]
+    (v: T#WrappedType, c: Constant[T])(implicit t: RType[T#WrappedType]) = {
+    // for any Byte and Short value `v`, lifting of array should succeed
```
Comment (aslesarenko) [inline interpreter/shared/src/test/scala/sigma/ast/SigmaBuilderTest.scala:127]: Isn't is generic enough to work for any type? Comment somewhat misleading.

File: interpreter/shared/src/test/scala/sigma/ast/SigmaBuilderTest.scala [hunk 4]
```diff
@@ -152,40 +167,65 @@ class SigmaBuilderTest extends AnyPropSpec with ScalaCheckPropertyChecks with Ma
...
   }
   
   property("liftToConstant String") {
     val v = "abc"
     val c = StringConstant(v)
-    test[SString.type](v, c)
-    testArray[SString.type](v, c)    // TODO v6.0: String should be liftable at all (not supported in ErgoTree) (see https://github.com/ScorexFoundation/sigmastate-interpreter/issues/905)
-    testColl[SString.type](v, c)
+    if (!VersionContext.current.isV6SoftForkActivated) {
+      // v6.0: String should be liftable at all (not supported in ErgoTree) (see https://github.com/ScorexFoundation/sigmastate-interpreter/issues/905)
```
Comment (aslesarenko) [inline interpreter/shared/src/test/scala/sigma/ast/SigmaBuilderTest.scala:206]:
```suggestion
      // v6.0: String should not be liftable at all (not supported in ErgoTree) (see https://github.com/ScorexFoundation/sigmastate-interpreter/issues/905)
```

**Review by aslesarenko — APPROVED** [review]
LGTM, but please see comments

File: interpreter/shared/src/test/scala/sigma/ast/SigmaBuilderTest.scala (hunk 3 above)
Comment (kushti) [reply interpreter/shared/src/test/scala/sigma/ast/SigmaBuilderTest.scala]: fixed

File: data/js/src/main/scala/sigma/Platform.scala (hunk 1 above)
Comment (kushti) [reply data/js/src/main/scala/sigma/Platform.scala:35]: fixed

File: data/jvm/src/main/scala/sigma/Platform.scala (hunk 2 above)
Comment (kushti) [reply data/jvm/src/main/scala/sigma/Platform.scala:34]: fixed

**Pull request merged**